### ISSUE TITLE
Fails on creating encrypted directory

### ISSUE THREAD

**Issue by Brando753** [sovereign/sovereign#665]
`➜  sovereign git:(master) ansible-playbook -i hosts site.yml 

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
fatal: [107.***.***.***]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Shared connection to 107.***.***.*** closed.\r\n", "module_stdout": "sudo: a password is required\r\n", "msg": "MODULE FAILURE"}
	to retry, use: --limit @/home/brandon/workspace/sovereign/site.retry

PLAY RECAP *********************************************************************
107.***.***.***             : ok=0    changed=0    unreachable=0    failed=1   

➜  sovereign git:(master) ansible-playbook -i hosts site.yml --ask-sudo-pass
SUDO password: 

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [107.***.***.***]

TASK [common : Update apt cache] ***********************************************
changed: [107.***.***.***]

TASK [common : Upgrade all safe packages] **************************************
changed: [107.***.***.***]

TASK [common : Install necessities and nice-to-haves] **************************
changed: [107.***.***.***] => (item=[u'apache2', u'apt-transport-https', u'apticron', u'build-essential', u'debian-goodies', u'git', u'htop', u'iftop', u'iotop', u'molly-guard', u'mosh', u'python-software-properties', u'ruby', u'screen', u'sudo', u'unattended-upgrades', u'vim', u'zsh'])

TASK [common : timezone - configure /etc/timezone] *****************************
changed: [107.***.***.***]

TASK [common : timezone - Set localtime to UTC] ********************************
ok: [107.***.***.***]

TASK [common : timezone - reconfigure tzdata] **********************************
changed: [107.***.***.***]

TASK [common : Apticron email configuration] ***********************************
changed: [107.***.***.***]

TASK [common : Create decrypted directory (even if encfs isn't used)] **********
changed: [107.***.***.***]

TASK [common : Set decrypted directory permissions] ****************************
changed: [107.***.***.***]

TASK [common : Ensure locale en_US.UTF-8 locale is present] ********************
ok: [107.***.***.***]

TASK [common : Install encfs & fuse] *******************************************
changed: [107.***.***.***] => (item=[u'encfs', u'fuse', u'libfuse-dev'])

TASK [common : Create encrypted directory] *************************************
changed: [107.***.***.***]

TASK [common : Check if the /encrypted directory is empty] *********************
fatal: [107.***.***.***]: FAILED! => {"changed": false, "cmd": "ls /encrypted/*", "delta": "0:00:00.002053", "end": "2017-03-16 10:08:31.122902", "failed": true, "rc": 2, "start": "2017-03-16 10:08:31.120849", "stderr": "ls: cannot access /encrypted/*: No such file or directory", "stdout": "", "stdout_lines": [], "warnings": []}
...ignoring

TASK [common : If /encrypted is empty, create the encfs there] *****************
fatal: [107.***.***.***]: FAILED! => {"changed": true, "cmd": "printf \"p\\nALucgc0VVm1RhOg0MFlN\" | encfs /encrypted /decrypted --public --stdinpass && touch /decrypted/test", "delta": "0:00:03.801528", "end": "2017-03-16 10:08:35.624470", "failed": true, "rc": 1, "start": "2017-03-16 10:08:31.822942", "stderr": "fuse: device not found, try 'modprobe fuse' first\nfuse failed.  Common problems:\n - fuse kernel module not installed (modprobe fuse)\n - invalid options -- see usage message", "stdout": "Creating new encrypted volume.\nPlease choose from one of the following options:\n enter \"x\" for expert configuration mode,\n enter \"p\" for pre-configured paranoia mode,\n anything else, or an empty line will select standard mode.\n?> \nParanoia configuration selected.\n\nConfiguration finished.  The filesystem to be created has\nthe following properties:\nFilesystem cipher: \"ssl/aes\", version 3:0:2\nFilename encoding: \"nameio/block\", version 3:0:1\nKey Size: 256 bits\nBlock Size: 1024 bytes, including 8 byte MAC header\nEach file contains 8 byte header with unique IV data.\nFilenames encoded using IV chaining mode.\nFile data IV is chained to filename IV.\nFile holes passed through to ciphertext.\n\n-------------------------- WARNING --------------------------\nThe external initialization-vector chaining option has been\nenabled.  This option disables the use of hard links on the\nfilesystem. Without hard links, some programs may not work.\nThe programs 'mutt' and 'procmail' are known to fail.  For\nmore information, please see the encfs mailing list.\nIf you would like to choose another configuration setting,\nplease press CTRL-C now to abort and start over.\n\nNow you will need to enter a password for your filesystem.\nYou will need to remember this password, as there is absolutely\nno recovery mechanism.  However, the password can be changed\nlater using encfsctl.", "stdout_lines": ["Creating new encrypted volume.", "Please choose from one of the following options:", " enter \"x\" for expert configuration mode,", " enter \"p\" for pre-configured paranoia mode,", " anything else, or an empty line will select standard mode.", "?> ", "Paranoia configuration selected.", "", "Configuration finished.  The filesystem to be created has", "the following properties:", "Filesystem cipher: \"ssl/aes\", version 3:0:2", "Filename encoding: \"nameio/block\", version 3:0:1", "Key Size: 256 bits", "Block Size: 1024 bytes, including 8 byte MAC header", "Each file contains 8 byte header with unique IV data.", "Filenames encoded using IV chaining mode.", "File data IV is chained to filename IV.", "File holes passed through to ciphertext.", "", "-------------------------- WARNING --------------------------", "The external initialization-vector chaining option has been", "enabled.  This option disables the use of hard links on the", "filesystem. Without hard links, some programs may not work.", "The programs 'mutt' and 'procmail' are known to fail.  For", "more information, please see the encfs mailing list.", "If you would like to choose another configuration setting,", "please press CTRL-C now to abort and start over.", "", "Now you will need to enter a password for your filesystem.", "You will need to remember this password, as there is absolutely", "no recovery mechanism.  However, the password can be changed", "later using encfsctl."], "warnings": []}
	to retry, use: --limit @/home/brandon/workspace/sovereign/site.retry

PLAY RECAP *********************************************************************
107.***.***.***             : ok=14   changed=10   unreachable=0    failed=1   

`

**Comment by cstich** [comment]
Can you try creating the encfs manually and see what happens?

**Comment by WMPayne** [comment]
I had a similar problem when I was installing. The shell command was ls /encrypted/* in common/tasks/encfs.yml and when I removed both the /* then it worked. The command being ls /encrypted

**Comment by jmcgnh** [comment]
Yes, I saw this, too. I was using the vagrant jessie environment with box-cutter/Debian8 (2.0.8) 

If you look at the first failure in the trace, regarding 'ls /encrypted/*' command, you'll see that this is a probe to see if the encrypted folder not only exists but has content beyond the (hidden) .xml configuration file. The error result is ignored but registered to switch between the two following actions.

The second failure in the trace contains the string 

> fuse: device not found, try 'modprobe fuse' first\nfuse failed.  Common problems:\n - fuse kernel module not installed (modprobe fuse)\n

Based on closed issue #468 and reports of success, I expected this not to still be happening. I did a reboot, checked 'modprobe fuse' and removed the /encrypted and /decrypted folders. When I restarted the provisioning, the playbook ran to completion and I had a populated /decrypted folder properly mounted on /encrypted.

So I'm a bit baffled. I'll probably go back to my [jmcgnh#9]( https://github.com/jmcgnh/sovereign/issues/9) patch for further testing, but mikeashley has pointed out that this approach won't work on systems that don't allow modprobe.

[And sorry for not responding on #468 - I've been away from this project for most of the past 18 months (!) and did not see the messages directed at me.]

**Comment by erlepereira** [comment]
on things like containers where modprobe won't work, as long as the fuse module is loaded into the kernel this could work:

`mknod -m 666 dev/fuse c 10 229`
once created, the scripts ran through as needed. Maybe a shell command to create and re-attempt somehow?

I can verify this worked for me, LXC + Debian Jessie, currently experimenting with it myself.

mentioned here : https://github.com/lxc/lxc/issues/80#issuecomment-27190998
